### PR TITLE
lnms snmp:fetch command and new SNMP code

### PR DESCRIPTION
**Pull request by murrant**
`lnms snmp:fetch localhost SNMPv2-MIB::sysDescr.0`

Makes it easy to call snmp commands without specifying credentials and mibs.

Uses new snmp code.

Splits fetching and parsing code apart.  This way you can get the data as desire, or parse the raw output.
-OXQ is required for parsing.  Mostly using defaults will get what you want (including defaulting to the primary device):

`$sysDescr = NetSnmp::get('SNMPv2-MIB::sysDescr.0')->value();`
`$ifAlias = NetSnmp::walk('IF-MIB::ifAlias')->values();`
`$ifTabel = NetSnmp::walk('IF-MIB::ifTable')->table();`

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: LibreNMS/Data/Source/SnmpResponse.php [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    public function __construct(string $output, int $exitCode = 0)
+    {
+        $this->exitCode = $exitCode;
+        $this->raw = preg_replace('/Wrong Type \(should be .*\): /', '', $output);
+    }
+
+    public function isValid(): bool
+    {
+        $invalid = empty($this->raw)
+//            || $this->exitCode !== 0
```
Comment (Jellyfrog) [inline LibreNMS/Data/Source/SnmpResponse.php:52]: remove?

File: LibreNMS/Data/Source/SnmpResponse.php [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            || preg_match('/(No Such Instance|No Such Object|No more variables left|Authentication failure|Timeout: No Response from)/i', $this->raw, $matches);
```
Comment (Jellyfrog) [inline LibreNMS/Data/Source/SnmpResponse.php:53]: Very unlikely, but a response coooooould match this.
It's not printed on stderr or so instead?
Or maybe can restrict it to beginning of line or something?

File: LibreNMS/Data/Source/SnmpResponse.php [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    {
+        return Arr::first($this->values(), null, '');
+    }
+
+    public function values(): array
+    {
+        $values = [];
+        $line = strtok($this->raw, PHP_EOL);
+        while ($line !== false) {
+            if (Str::contains($line, 'at this OID') || Str::contains($line, 'this MIB View')) {
```
Comment (Jellyfrog) [inline LibreNMS/Data/Source/SnmpResponse.php:82]: deserves a comment

File: LibreNMS/Data/Source/SnmpResponse.php [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                $line = strtok(PHP_EOL);
+                continue;
+            }
+
+            [$oid, $value] = explode(' = ', $line, 2);
+
+            $line = strtok(PHP_EOL); // get the next line and concatenate multi-line values
+            while ($line !== false && ! Str::contains($line, '=')) {
```
Comment (Jellyfrog) [inline LibreNMS/Data/Source/SnmpResponse.php:90]: shouldn't this also be ` = ` (with spaces) ?
In that case, maybe make it a `const DELIMITER` or so?

**Comment by Jellyfrog** [comment]
Amazing work like always!
I think the output parsing deserves some test cases

File: LibreNMS/Data/Source/SnmpResponse.php (hunk 2 above)
Comment (murrant) [reply LibreNMS/Data/Source/SnmpResponse.php:53]: This is copied from the existing code.
I think we could remove the case insensitive switch though.

File: LibreNMS/Data/Source/SnmpResponse.php (hunk 1 above)
Comment (murrant) [reply LibreNMS/Data/Source/SnmpResponse.php:52]: Yeah, I'm still on the fence about it.

**Comment by murrant** [comment]
Yeah, I should write tests :)

**Comment by murrant** [comment]
LGTM :)

**Review by Jellyfrog — APPROVED** [review]

**Pull request merged**